aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequestFactory.java

```diff
@@ -230,19 +239,60 @@ private AppSyncGraphQLRequestFactory() { }
 
             doc.append("subscription ")
                     .append(StringUtils.allCapsToPascalCase(typeStr))
-                    .append(StringUtils.capitalizeFirst(graphQlTypeName))
-                    .append("{")
+                    .append(StringUtils.capitalizeFirst(graphQlTypeName));
+
+            if (schema.hasOwnerAuthorization()) {
+                doc.append("($owner: String!) ");
+            }
+
+            doc.append("{")
                     .append(StringUtils.allCapsToCamelCase(typeStr))
-                    .append(StringUtils.capitalizeFirst(graphQlTypeName))
-                    .append("{")
-                    .append(getModelFields(modelClass, DEFAULT_LEVEL_DEPTH))
-                    .append("}}");
+                    .append(StringUtils.capitalizeFirst(graphQlTypeName));
 
-            return new GraphQLRequest<>(
-                    doc.toString(),
-                    modelClass,
-                    new GsonVariablesSerializer()
-            );
+            if (schema.hasOwnerAuthorization()) {
+                doc.append("(owner: $owner) ");
+            }
+
+            doc.append("{").append(getModelFields(modelClass, DEFAULT_LEVEL_DEPTH));
+
+            if (schema.hasOwnerAuthorization()) {
+                doc.append(" owner ");
+            }
+
+            doc.append("}}");
+
+            if (schema.hasOwnerAuthorization()) {
+                if (cognitoAuth == null) {
+                    throw new ApiException(
+                        "Attempted to subscribe to a model with owner based authorization without a Cognito provider",
+                        "Did you initialize AWSMobileClient before making this call?"
+                    );
+                }
+
+                String username = cognitoAuth.getUsername();
+
+                if (username == null) {
+                    throw new ApiException(
+                            "Attempted to subscribe to a model with owner based authorization without a username",
+                            "Make sure that a user is logged in before subscribing to a model with owner based auth"
+                    );
+                }
+
+                return new GraphQLRequest<>(
+                        doc.toString(),
+                        Collections.singletonMap("owner", username),
```

Comment:
`Immutable.of(Collections.singletonMap("owner", username))` if the map can be immutable.

Reply:
Done